src/index.ts

```diff
@@ -67,42 +68,59 @@ extendEnvironment((hre) => {
   })
 })
 
-export const TASK_PREPARE_ARTIFACTS = "prepare-artifacts"
+extendConfig(
+  (config: HardhatConfig, userConfig: Readonly<HardhatUserConfig>) => {
+    const exportUserConfig = userConfig.deploymentArtifactsExport
 
-task(TASK_PREPARE_ARTIFACTS)
-  .addOptionalPositionalParam(
-    "destination",
-    "destination folder where the artifacts files will be written to",
-    "artifacts",
-    types.string
-  )
-  .setAction(async (args, hre) => {
-    const sourceDir = path.join(hre.config.paths.deployments, hre.network.name)
-    const destinationDir = path.join(hre.config.paths.root, args.destination)
-
-    console.log(
-      `Preparing deployment artifacts for network ${hre.network.name}...`
-    )
-
-    if (!fs.pathExistsSync(sourceDir)) {
-      throw new HardhatPluginError(
-        "@keep-network/hardhat-helpers",
-        `source deployments artifacts directory doesn't exist [${sourceDir}]`
+    if (config.deploymentArtifactsExport === undefined) {
+      config.deploymentArtifactsExport = {}
+    }
+
+    let defaultDestinationDir = exportUserConfig?.default ?? "artifacts"
+
+    if (path.isAbsolute(defaultDestinationDir)) {
+      defaultDestinationDir = defaultDestinationDir
+    } else {
+      defaultDestinationDir = path.normalize(
+        path.join(config.paths.root, defaultDestinationDir)
       )
     }
 
-    console.debug(
-      `  Source:      ${sourceDir}\n  Destination: ${destinationDir}`
-    )
+    const networks = Object.keys(config.networks)
 
-    fs.ensureDirSync(destinationDir)
-    fs.emptyDirSync(destinationDir)
+    networks.forEach((value: string) => {
+      if (
+        exportUserConfig === undefined ||
+        exportUserConfig[value] === undefined
+      ) {
+        config.deploymentArtifactsExport[value] = defaultDestinationDir
+      } else {
+        let networkDestinationDir = exportUserConfig[value]!
```

Comment:
I'm not following "networkDestinationDir". We define it here, then we assign a different value to it in the `if/else`, and then we do nothing.

Reply:
Good catch! It was a bug. Fixed in 531c2c5c3dde9621f2ab67681b43cf859e917f19